modules/acquia_lift_publisher/src/EventSubscriber/Publish/PublishOnlyRendered.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace Drupal\acquia_lift_publisher\EventSubscriber\Publish;
+
+use Acquia\ContentHubClient\CDF\CDFObject;
+use Drupal\acquia_contenthub\AcquiaContentHubEvents;
+use Drupal\acquia_contenthub\ContentHubCommonActions;
+use Drupal\acquia_contenthub\Event\PrunePublishCdfEntitiesEvent;
+use Drupal\acquia_contenthub_publisher\ContentHubPublisherEvents;
+use Drupal\acquia_contenthub_publisher\Event\ContentHubEntityEligibilityEvent;
+use Drupal\acquia_lift_publisher\Form\AcquiaLiftPublisherSettingsForm;
+use Drupal\Core\Config\ImmutableConfig;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Responsible for filtering contents based on publishing configurations.
+ *
+ * @see \Drupal\acquia_lift_publisher\Form\AcquiaLiftPublisherSettingsForm::$pushSettingField
+ * @package Drupal\acquia_lift_publisher\PruneCdf
+ */
+class PublishOnlyRendered implements EventSubscriberInterface {
+
+  /**
+   * Content Hub common actions.
+   *
+   * @var \Drupal\acquia_contenthub\ContentHubCommonActions
+   */
+  protected $commonActions;
+
+  /**
+   * Acquia lift publisher configuration object.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   * @see \Drupal\acquia_lift_publisher\Form\AcquiaLiftPublisherSettingsForm
+   */
+  protected $publisherSettings;
+
+  /**
+   * PublishOnlyRendered constructor.
+   *
+   * @param \Drupal\acquia_contenthub\ContentHubCommonActions $common_actions
+   *   The Content Hub common actions service.
+   * @param \Drupal\Core\Config\ImmutableConfig $config
+   *   The publisher configuration object.
+   */
+  public function __construct(ContentHubCommonActions $common_actions, ImmutableConfig $config) {
+    $this->commonActions = $common_actions;
+    $this->publisherSettings = $config;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      ContentHubPublisherEvents::ENQUEUE_CANDIDATE_ENTITY => ['onEnqueueCandidateEntity', 100],
+      AcquiaContentHubEvents::PRUNE_PUBLISH_CDF_ENTITIES => ['onPrunePublishCdfEntities', 100],
+    ];
+  }
+
+  /**
+   * Checks if the entity in hand is a renderable entity.
+   *
+   * Only enqueue the entity if it is renderable.
+   *
+   * @param \Drupal\acquia_contenthub_publisher\Event\ContentHubEntityEligibilityEvent $event
+   *   The current enqueue eligibility event.
+   *
+   * @throws \Exception
+   */
+  public function onEnqueueCandidateEntity(ContentHubEntityEligibilityEvent $event): void {
+    if (!$this->personalizedContentPushIsActive()) {
+      return;
+    }
+
+    $entity = $event->getEntity();
+    $cdfs = $this->commonActions->getEntityCdf($entity);
+    foreach ($cdfs as $cdf) {
+      if ($cdf->getType() !== 'rendered_entity') {
```

Comment:
We should double-check that this is enough. @ynx , do you know if this will fulfill the taxonomy terms needed for content recommendations?